File: Tutorials/WinForms/Classic/CS/MainForm.cs
using System;
using DevExpress.XtraBars.Docking2010.Views;
using DevExpress.XtraBars;
using DevExpress.XtraBars.Navigation;
using DevExpress.XtraBars.Ribbon;
using System.Linq;

namespace WinFormsApplication {
    public partial class MainForm : RibbonForm {
        public MainForm() {
            InitializeComponent();
            tabbedView.DocumentActivated += TabbedView_DocumentActivated;
            tabbedView.DocumentClosed += TabbedView_DocumentClosed;
            tabbedView.QueryControl += TabbedView_QueryControl;
            customersAccordionControlElement.Tag = nameof(CustomersListForm);
            ordersAccordionControlElement.Tag = nameof(OrdersListForm);
            ActivateDocument("Customers", nameof(CustomersListForm));
        }

        private void TabbedView_QueryControl(object sender, DevExpress.XtraBars.Docking2010.Views.QueryControlEventArgs e) {
            switch (e.Document.ControlName) {
                case nameof(CustomersListForm):
                    e.Control = new CustomersListForm();
                    break;
                case nameof(OrdersListForm):
                    e.Control = new OrdersListForm();
                    break;
                default:
                    throw new ArgumentException($"Unknown control name {e.Document.ControlName}");
            }
        }

        private void TabbedView_DocumentActivated(object sender, DocumentEventArgs e) {
            accordionControl.SelectedElement = accordionControl.GetElements()
                .Single(t => (string)t.Tag == e.Document.ControlName);
            if (ribbonControl.MergedPages.Count > 0) {
                ribbonControl.SelectedPage = ribbonControl.MergedPages[0];
            }
        }

        private void TabbedView_DocumentClosed(object sender, DocumentEventArgs e) {
            if(tabbedView.Documents.Count == 0) {
                accordionControl.SelectedElement = null;
            }
        }
        void ActivateDocument(string caption, string controlName) {
            BaseDocument document = tabbedView.Documents.FindFirst(d => d.ControlName == controlName);
            if (document == null)
                document = tabbedView.AddDocument(caption, controlName);
            tabbedView.Controller.Activate(document);
        }

        private void customersAccordionControlElement_Click(object sender, EventArgs e) {
            ActivateDocument("Customers", nameof(CustomersListForm));
        }

        private void ordersAccordionControlElement_Click(object sender, EventArgs e) {
            ActivateDocument("Orders", nameof(OrdersListForm));
        }

        private void customersBarButtonItem_ItemClick(object sender, ItemClickEventArgs e) {
            ActivateDocument("Customers", nameof(CustomersListForm));
        }

        private void ordersBarButtonItem_ItemClick(object sender, ItemClickEventArgs e) {
            ActivateDocument("Orders", nameof(OrdersListForm));
        }
    }
}

File: Tutorials/WinForms/Classic/VB/MainForm.vb
Imports System
Imports DevExpress.XtraBars.Docking2010.Views
Imports DevExpress.XtraBars
Imports DevExpress.XtraBars.Navigation
Imports DevExpress.XtraBars.Ribbon
Imports System.Linq
Imports WinFormsApplication.XpoTutorial
Imports DevExpress.Xpo

Partial Public Class MainForm
    Inherits RibbonForm

    Public Sub New()
        ConnectionHelper.Connect(False)
        Using uow As New UnitOfWork()
            DemoDataHelper.Seed(uow)
        End Using
        InitializeComponent()
        AddHandler tabbedView.DocumentActivated, AddressOf TabbedView_DocumentActivated
        AddHandler tabbedView.DocumentClosed, AddressOf TabbedView_DocumentClosed
        AddHandler tabbedView.QueryControl, AddressOf TabbedView_QueryControl
        customersAccordionControlElement.Tag = NameOf(CustomersListForm)
        ordersAccordionControlElement.Tag = NameOf(OrdersListForm)
        ActivateDocument("Customers", NameOf(CustomersListForm))
    End Sub

    Private Sub TabbedView_QueryControl(ByVal sender As Object, ByVal e As DevExpress.XtraBars.Docking2010.Views.QueryControlEventArgs)
        Select Case e.Document.ControlName
            Case NameOf(CustomersListForm)
                e.Control = New CustomersListForm()
            Case NameOf(OrdersListForm)
                e.Control = New OrdersListForm()
            Case Else
                Throw New ArgumentException($"Unknown control name {e.Document.ControlName}")
        End Select
    End Sub

    Private Sub TabbedView_DocumentActivated(ByVal sender As Object, ByVal e As DocumentEventArgs)
        accordionControl.SelectedElement = accordionControl.GetElements().Single(Function(t) CStr(t.Tag) = e.Document.ControlName)
        If ribbonControl.MergedPages.Count > 0 Then
            ribbonControl.SelectedPage = ribbonControl.MergedPages(0)
        End If
    End Sub

    Private Sub TabbedView_DocumentClosed(ByVal sender As Object, ByVal e As DocumentEventArgs)
        If tabbedView.Documents.Count = 0 Then
            accordionControl.SelectedElement = Nothing
        End If
    End Sub
    Private Sub ActivateDocument(ByVal caption As String, ByVal controlName As String)
        Dim document As BaseDocument = tabbedView.Documents.FindFirst(Function(d) d.ControlName = controlName)
        If document Is Nothing Then
            document = tabbedView.AddDocument(caption, controlName)
        End If
        tabbedView.Controller.Activate(document)
    End Sub

    Private Sub customersAccordionControlElement_Click(ByVal sender As Object, ByVal e As EventArgs) Handles customersAccordionControlElement.Click
        ActivateDocument("Customers", NameOf(CustomersListForm))
    End Sub

    Private Sub ordersAccordionControlElement_Click(ByVal sender As Object, ByVal e As EventArgs) Handles ordersAccordionControlElement.Click
        ActivateDocument("Orders", NameOf(OrdersListForm))
    End Sub

    Private Sub customersBarButtonItem_ItemClick(ByVal sender As Object, ByVal e As ItemClickEventArgs) Handles customersBarButtonItem.ItemClick
        ActivateDocument("Customers", NameOf(CustomersListForm))
    End Sub

    Private Sub ordersBarButtonItem_ItemClick(ByVal sender As Object, ByVal e As ItemClickEventArgs) Handles ordersBarButtonItem.ItemClick
        ActivateDocument("Orders", NameOf(OrdersListForm))
    End Sub
End Class